pkg/generic/httpthrift_codec.go

```diff
@@ -46,15 +51,32 @@ type HTTPRequest = descriptor.HTTPRequest
 type HTTPResponse = descriptor.HTTPResponse
 
 type httpThriftCodec struct {
-	svcDsc           atomic.Value // *idl
-	provider         DescriptorProvider
-	codec            remote.PayloadCodec
-	binaryWithBase64 bool
+	svcDsc              atomic.Value // *idl
+	provider            DescriptorProvider
+	codec               remote.PayloadCodec
+	binaryWithBase64    bool
+	enableDynamicgoReq  bool
+	enableDynamicgoResp bool
+	convOpts            conv.Options
 }
 
-func newHTTPThriftCodec(p DescriptorProvider, codec remote.PayloadCodec) (*httpThriftCodec, error) {
+func newHTTPThriftCodec(p DescriptorProvider, codec remote.PayloadCodec, opts ...DynamicgoOptions) (*httpThriftCodec, error) {
 	svc := <-p.Provide()
-	c := &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: false}
+	var c *httpThriftCodec
+	if opts == nil {
+		c = &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: false}
+	} else {
+		// codec with dynamicgo
+		convOpts := opts[0].ConvOpts
+		if !convOpts.EnableHttpMapping {
+			convOpts.EnableHttpMapping = true
+		}
+		binaryWithBase64 := false
+		if !convOpts.NoBase64Binary {
```

Comment:
@AsterDY binaryWithBase64 field (in original implementation) should be false by default. (You can refer to [https://github.com/cloudwego/kitex/pull/293/files](https://github.com/cloudwego/kitex/pull/293/files))
So when NoBase64Binary is set as false (or users do not set anything), binaryWithBase64 should be changed true in case of fallback.
It should be noted that the NoBase64Binary field must be true by default, but users may forget to set it. We can't recognize this case with this implementation...